syntax = "proto3";

package jinja_server;
option go_package = "/jinja2_server";

message StringsJob {
  string vars = 1;
  repeated string search_dirs = 2;
  repeated string templates = 3;
}

message FilesJob {
  string vars = 1;
  repeated string search_dirs = 2;
  repeated string templates = 3;
}

message SingleResult {
  optional string result = 1;
  optional string error = 2;
}

message JobResult {
  optional string error = 1;
  repeated SingleResult results = 2;
}

service Jinja2Server {
  rpc RenderStrings(StringsJob) returns(JobResult);
  rpc RenderFiles(FilesJob) returns(JobResult);
}
